controllers/auth.controller.js

```diff
@@ -1,17 +1,43 @@
-const authService = require("../services/auth.service");
+const router = require("express").Router();
+const User = require("../models/user.model");
+const jwt = require("jsonwebtoken");
+const AuthService = require("../services/auth.service")
 
 const responseWithToken = (res, data) => {
   return res.header("X-Auth-Token", data.token).json(data);
 };
 
-exports.login = (req, res, next) => {
-  try {
-    const data = await authService.login(req.body); // TODO probably broken - its just an example
-    return responseWithToken(res, data); //TODO
-  } catch (error) {
+exports.register = async (req, res, next) =>{
+  try{
+    const token = AuthService.register(req.body);
+    res.set("x-auth-token", token).json({message: "Tranks for registering"});
```

Comment:
res.header("X-Auth-Token", data.token).json(data);

where data is an object containing user data:
token, 
isAdmin (from user model)
name (from user model)

Reply:
_id